store/sqlstore/upgrade.go

```diff
@@ -818,8 +819,73 @@ func upgradeDatabaseToVersion525(sqlStore SqlStore) {
 func upgradeDatabaseToVersion526(sqlStore SqlStore) {
 	// TODO: uncomment when the time arrive to upgrade the DB for 5.26
 	//if shouldPerformUpgrade(sqlStore, VERSION_5_25_0, VERSION_5_26_0) {
+	if err := precheckMigrationToVersion526(sqlStore); err != nil {
+		mlog.Error("Error upgrading DB schema to 5.26.0", mlog.Err(err))
+		os.Exit(EXIT_GENERIC_FAILURE)
+	}
+
+	sqlStore.AlterColumnTypeIfExists("Teams", "Type", "VARCHAR(255)", "VARCHAR(255)")
+	sqlStore.AlterColumnTypeIfExists("Teams", "SchemeId", "VARCHAR(26)", "VARCHAR(26)")
+	sqlStore.AlterColumnTypeIfExists("IncomingWebhooks", "Username", "varchar(255)", "varchar(255)")
+	sqlStore.AlterColumnTypeIfExists("IncomingWebhooks", "IconURL", "text", "varchar(1024)")
+
 	sqlStore.CreateColumnIfNotExists("Sessions", "ExpiredNotify", "boolean", "boolean", "0")
 
 	//saveSchemaVersion(sqlStore, VERSION_5_26_0)
 	//}
 }
+
+func precheckMigrationToVersion526(sqlStore SqlStore) error {
+	teamsQuery, _, err := sqlStore.getQueryBuilder().Select(`CASE
+				WHEN CHAR_LENGTH(SchemeId) > 26 THEN FALSE
+				ELSE TRUE
+			END as schemeidok,
+			CASE
+				WHEN CHAR_LENGTH(Type) > 255 THEN FALSE
+				ELSE TRUE
+			END as typeok`).
+		From("Teams").ToSql()
+	if err != nil {
+		return err
+	}
+	webhooksQuery, _, err := sqlStore.getQueryBuilder().Select(`CASE
+				WHEN CHAR_LENGTH(Username) > 255 THEN FALSE
+				ELSE TRUE
+			END as usernameok,
+			CASE
+				WHEN CHAR_LENGTH(IconURL) > 1024 THEN FALSE
+				ELSE TRUE
+			END as iconurlok`).
+		From("IncomingWebhooks").ToSql()
+	if err != nil {
+		return err
+	}
+
+	var schemeIdOk bool
+	var typeOk bool
+	row := sqlStore.GetMaster().Db.QueryRow(teamsQuery)
```

Comment:
Does this only check the first row of the result? Do we need to check every row?

Reply:
Yes you’re absolutely right! I wanted to aggregate but I made the mistake of think about the select clause as an aggregation one 🤦. I’ll change it to an aggregate one, thank you!